pkg/controller/disruption/disruption.go

```diff
@@ -595,11 +595,18 @@ func (dc *DisruptionController) trySync(pdb *policy.PodDisruptionBudget) error {
 		dc.recorder.Eventf(pdb, v1.EventTypeNormal, "NoPods", "No matching pods found")
 	}
 
-	expectedCount, desiredHealthy, err := dc.getExpectedPodCount(pdb, pods)
+	expectedCount, desiredHealthy, nakedPods, err := dc.getExpectedPodCount(pdb, pods)
 	if err != nil {
 		dc.recorder.Eventf(pdb, v1.EventTypeWarning, "CalculateExpectedPodCountFailed", "Failed to calculate the number of expected pods: %v", err)
 		return err
 	}
+	// We have naked pods, instead of erroring and hotlooping in disruption controller, enqueue and see if the
+	// error goes away. To identify the pods, we're emitting an event.
+	if len(nakedPods) > 0 {
+		dc.recorder.Eventf(pdb, v1.EventTypeWarning, "NakedPodsFound",
+			"Found naked pods associated with this PDB: %v", strings.Join(nakedPods, ",'"))
+		dc.enqueuePdb(pdb)
```

Comment:
I think this needs an e2e test that verifies the failure doesn't happen anymore.  Is enqueue going to do backoff and a delay?

Note that it is very easy for multiple controllers to conflict on pods, so they won't ever go away.  So if this is just another version of the hotloop that's not a full fix.

Reply:
> I think this needs an e2e test that verifies the failure doesn't happen anymore

I added a unit test for the same before this patch we'd get an error, now we don't get it anymore.

Enqueue doesn't do a delayed backoff. But we are using DefaultControllerRateLimiter() helps in ratelimiting. I am not sure, if adding constant delay would help considering the status wouldn't have changed but I am ambivalent on it.

> Note that it is very easy for multiple controllers to conflict on pods, so they won't ever go away. So if this is just another version of the hotloop that's not a full fix.

You mean continuously sending events? How else can we communicate that there are naked pods having PDB. Either we need to emit events or log and leave. I am fine with a higher level log message but I don't know if it'd be sufficient.

Reply:
When PDB controller detects a pod covered by a PDB that doesn't have a controller, it needs to not generate endless events for it.  You've just moved "writing status continuously" (very bad) to "writing events continuously" (bad).  We need to move to "write one status or one event that summarizes that there are unmanaged pods, until such a time as the change happens".

You should not send one event per enqueue, you should send one event or do one status write per "pdb has unmanaged pods" (regardless of how many unmanaged pods there are, or whether they change).

Reply:
As discussed offline, `found no controller ref for pod abc` is not actionable, since:

 https://kubernetes.io/docs/tasks/run-application/configure-pdb/#arbitrary-controllers-and-selectors clearly says we can use PDB with bare pods:
`You can use a PDB with pods controlled by another type of controller, by an "operator", or bare pods, but with these restrictions:`

Technically, there is no point in erroring and updating status if there is an unmanaged pod, so I am switching it to log instead of emitting.